pkg/cli/tail.go

```diff
@@ -222,24 +275,46 @@ func (c *TailCommand) Run(ctx context.Context, args []string) error {
 	return nil
 }
 
-func (c *TailCommand) tail(ctx context.Context) ([]*loggingpb.LogEntry, error) {
-	var p logPuller
-	if c.testPuller != nil {
-		p = c.testPuller
-	} else {
-		logClient, err := logging.NewClient(ctx)
+func (c *TailCommand) stream(ctx context.Context, extra []validation.Validator, puller logPuller) error {
+	// logCh is used to receive logEntriesfrom StreamPull
+	logCh := make(chan *loggingpb.LogEntry)
+
+	var failCount, totalCount int
+
+	var perr error
+	go func() {
+		if err := puller.StreamPull(ctx, c.queryFilter(), logCh); err != nil {
+			perr = fmt.Errorf("StreamPull failed: %w", err)
+		}
+		close(logCh)
+	}()
+
+	for l := range logCh {
+		totalCount++
+		js, err := protojson.Marshal(l)
```

Comment:
Should we wait for a second for each read?

Reply:
@yolocs what do you think?

Reply:
Let's skip it for now.